From ddcd59e33890e81aedcecca6a42e67a0f757b040 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:07 -0400
Subject: [fs] nfsd: Using path_get when assigning path for export

Message-id: <1410898849-22126-4-git-send-email-bfields@redhat.com>
Patchwork-id: 95691
O-Subject: [Patch RHEL7 003/165] NFSD: Using path_get when assigning path for export
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 0da22a919d6972f629407f79fc096f29d23a4942
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/export.c b/fs/nfsd/export.c
index 13b85f9..ef2d9d6 100644
--- a/fs/nfsd/export.c
+++ b/fs/nfsd/export.c
@@ -698,8 +698,8 @@ static void svc_export_init(struct cache_head *cnew, struct cache_head *citem)
 
  kref_get(&item->ex_client->ref);
  new->ex_client = item->ex_client;
- new->ex_path.dentry = dget(item->ex_path.dentry);
- new->ex_path.mnt = mntget(item->ex_path.mnt);
+ new->ex_path = item->ex_path;
+ path_get(&item->ex_path);
  new->ex_fslocs.locations = NULL;
  new->ex_fslocs.locations_count = 0;
  new->ex_fslocs.migrated = 0;
-- 
1.7.1